### DISCUSSION TITLE
git copilot not working in windows vs code as well in pycharm

### DISCUSSION THREAD

**Discussion by raushan80-443** [community/community#12429] [Copilot Conversations] [answered]
earlier I used linux to use co pilot and it was working just awesom but I am unable to use it in windows 
<img width="633" alt="gitco" src="https://user-images.githubusercontent.com/73380685/156866300-9dc76d3c-038c-479c-a7fe-293eea72bac1.png">
any one please help me

**Comment by Ashwin-droid** [comment]
go to github account settings > applications >authorized github apps >github copilot plugin > revoke

then reinstall the extension and log in

**Reply by raushan80-443** [reply]
nothing happens same response as earlier but the error messege in vs code is
[INFO] [auth] [2022-03-05T23:23:24.759Z] Invalid copilot token: invalid expires_at:1646502859 now:1646522604
[ERROR] [default] [2022-03-05T23:23:24.762Z] GitHub Copilot could not connect to server. Extension activation failed: "Failed to get copilot token"

**Reply by Ashwin-droid** [reply]
this was supposed to be for IntelliJ pycharm
is your vscode up to date (and also legit) - this is required as there are very recent fixes for just copilot 
also is the plugin legit installed from https://marketplace.visualstudio.com/items?itemName=GitHub.copilot and/or https://plugins.jetbrains.com/plugin/17718-github-copilot

**Reply by raushan80-443** [reply]
not working

**Reply by Ashwin-droid** [reply]
Ok fine last fix that I know of

Log out of vscode and log back in
Because if you see the log your api token is expired and this might help in renewing it

**Comment by davecheney** [comment]
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Comment by raushan80-443** [comment, marked as answer]
Hey I just restart my pc and it is working ,thank u all for you time
